components/src/widget/views/confirmation/confirmation.ts

```diff
@@ -267,15 +267,17 @@ export class PaymentConfirmation extends LitElement {
     debitAmount: Amount
     receiveAmount: Amount
   }): Promise<PendingGrant> {
-    const { apiUrl } = this.configController.config
+    const { apiUrl, frontendUrl } = this.configController.config
     const url = new URL('/tools/payment/grant', apiUrl).href
+    const redirectUrl = new URL('payment-confirmation', frontendUrl).href
 
     const response = await fetch(url, {
       method: 'POST',
       headers: {
         'Content-Type': 'application/json'
       },
       body: JSON.stringify({
+        redirectUrl,
```

Comment:
Just a thought, not related to CI - maybe it is going to be worth adding an error log for the bad request branch to get more context about the error (I do not know how the API is currently build but maybe more information can help for bug reports latern on).